utils/frame/remote-externalities/src/lib.rs

```diff
@@ -351,14 +351,57 @@ where
 		prefix: StorageKey,
 		at: B::Hash,
 	) -> Result<Vec<StorageKey>, &'static str> {
-		let mut last_key: Option<StorageKey> = None;
+		const NUM_PARTS: usize = 4; // number of parts to divide the prefix into
+
+		let mut tasks = Vec::with_capacity(NUM_PARTS);
+
+		// Divide the prefix into N parts
+		let mut start_key = prefix.clone();
+		let mut end_key = prefix.clone();
 		let mut all_keys: Vec<StorageKey> = vec![];
-		let keys = loop {
+
+		for _ in 0..NUM_PARTS {
+			end_key.0[0] = start_key.0[0].wrapping_add(1);
+			let task = self.rpc_get_keys_paged_range(start_key.clone(), end_key.clone(), at);
+			tasks.push(task);
+			start_key.0[0] = end_key.0[0].wrapping_add(1);
+		}
+
+		for task in futures::future::join_all(tasks).await {
+			let mut keys = task?;
+			all_keys.append(&mut keys);
+		}
+
+		// Filter out default child storage keys
+		let filtered_keys = all_keys
+			.into_iter()
+			.filter(|k| !is_default_child_storage_key(&k.0))
+			.collect::<Vec<_>>();
```

Comment:
why'd you add filtering here?

Reply:
Yes, It's not required. I completely overlooked the rpc_get_pairs_paged function.